src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/PackageSourceMappingUtility.cs

```diff
@@ -41,9 +43,12 @@ internal static class PackageSourceMappingUtility
 
             // Get all newly added package IDs that were not previously Source Mapped.
             // Always include the Package ID being installed since it takes precedence over any globbing.
-            string[] packageIdsNeedingNewSourceMappings = addedPackageIdsWithoutExistingMappings
-                .Append(userAction.PackageId)
-                .ToArray();
+            if (!topLevelPackageIdWasAdded)
+            {
+                addedPackageIdsWithoutExistingMappings.Add(userAction.PackageId);
+            }
+
+            string[] packageIdsNeedingNewSourceMappings = addedPackageIdsWithoutExistingMappings.ToArray();
```

Comment:
`.ToArray()` causes an allocation. Can you change `CreateAndSavePackageSouyrceMappings` to use ` List<string>` instead of `string[]`? It's private, and this assembly isn't part of our public API anyway, so the variable shouldn't need to be an array, unless it calls a public API later on that does need to be an array.

Reply:
Sure, I can go ahead and refactor that in this PR.